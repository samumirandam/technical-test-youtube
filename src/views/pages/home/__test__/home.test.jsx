/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/no-unresolved */
import React from 'react';

import { render, screen } from '@utils/test-utils';

import Home from '../index';

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Home {...setupStore} />, {});
};

describe('Test for Home page component', () => {
  test('Should render without errors', () => {
    setup();
    expect(screen.getByTestId('Home')).toBeTruthy();
  });

  test('Should renders the same component', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
