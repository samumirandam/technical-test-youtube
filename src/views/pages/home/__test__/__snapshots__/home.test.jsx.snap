// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Home page component Should renders the same component 1`] = `
<div>
  <section
    class="Home"
    data-testid="Home"
  >
    <h1
      class="Home__title"
    >
      Añadir nuevo video
    </h1>
    <div
      class="InputButton Home__input"
    >
      <input
        autocomplete="true"
        class="InputButton__input"
        placeholder="Añadir"
        type="text"
        value=""
      />
      <button
        class="Button InputButton__button primary secondary"
        type="button"
      >
        Añadir
      </button>
    </div>
    <section
      class="VideoList"
    />
  </section>
</div>
`;
